core/src/main/java/org/elasticsearch/search/internal/ShardSearchLocalRequest.java

```diff
@@ -167,6 +176,11 @@ protected void innerReadFrom(StreamInput in) throws IOException {
         source = in.readOptionalWriteable(SearchSourceBuilder::new);
         types = in.readStringArray();
         aliasFilter = new AliasFilter(in);
+        if (in.getVersion().onOrAfter(Version.V_5_2_0_UNRELEASED)) {
+            indexBoost = in.readFloat();
+        } else {
+            indexBoost = 1.0f;
```

Comment:
it would also be wise to unit test that newly added bw comp layer. We have something similar in ShardSearchTransportRequest for alias filters (we moved their parsing on the coord node with 5.1).